src/Carbon/CarbonPeriod.php

```diff
@@ -327,6 +327,15 @@ class CarbonPeriod extends DatePeriodBase implements Countable, JsonSerializable
 
     public function getIterator(): Generator
     {
+        // Safety check: if interval is empty AND has no step function, yield only start date and stop
+        // Note: Empty intervals with step functions (callables) are valid and should continue
+        if ($this->dateInterval->isEmpty() && $this->dateInterval->getStep() === null) {
+            $this->rewind();
+            yield 0 => $this->current();
```

Comment:
If we can demonstrate that an infinite loop is indeed possible (i.e. that it's possible to iterate more than `END_MAX_ATTEMPTS` times), then I don't think silently returning just 1 date is a correct way to handle this. It's putting the dirt under the carpet. I think it's way safer to throw an exception in such case so the user can discover that the interval is wrong and act upon it (not create such period then decide on their application what it should do instead).